File: 99_bottles.js
#!/usr/bin/env node

function bottles(n) {
  if (n == 0) { return };
  var affix = function(x) { return x == 1 ? "" : "s" };
  console.log(n + " bottle" + affix(n) + " of beer on the wall");
  console.log(n + " bottle" + affix(n) + " of beer");
  console.log("take one down, pass it around");
  console.log(n-1 + " bottle" + affix(n-1) + " of beer on the wall");
  bottles(n-1);
}

bottles(99);

File: 99_bottles.swift
#!/usr/bin/env swift
// http://rosettacode.org/wiki/99_Bottles_of_Beer

func bottles(n: Int) {
  guard n > 0 else { return }
  func affix(x: Int) -> String { return n == 1 ? "" : "s" }
  print("\(n) bottle\(affix(n)) of beer on the wall")
  print("\(n) bottle\(affix(n)) of beer")
  print("take one down, pass it around")
  print("\(n-1) bottle\(affix(n)) of beer on the wall")
  bottles(n-1)
}

bottles(99)
